### ISSUE TITLE
reportGeneralTypeIssues when subclassing TypedDict

### ISSUE THREAD

**Issue by abennett** [microsoft/pyright#4822]
**Describe the bug**
Pyright is reporting an error when simply subclassing TypedDict.

**To Reproduce**
Slap the code below in a file and check it.

**Expected behavior**
No errors.

**Screenshots or Code**
```python
from typing import TypedDict, List


class UserOrganization(TypedDict):
    id: str
    name: str
    slug: str
    creator: str | None
    collaborators_url: str
    projects_url: str
    created_at: str
    updated_at: str


UserOrganizationsResp = List[UserOrganization]
```

The above code throws:
```
/types.py:4:24 - error: Expected type expression but received "(typename: Unknown, fields: Unknown | None = None, /, *, total: bool = True, **kwargs: Unknown) -> Type[__class__TypedDictMeta]" (reportGeneralTypeIssues)
```
**VS Code extension or command-line**
Running `pyright` through nvim lsp.

**Comment by erictraut** [comment]
OK, let me know if you're able to repro. I'm going to close for now because I'm pretty confident this isn't a bug in pyright.

**Comment by abennett** [comment]
Still haven't been able to reliability reproduce in a couple different Darwin VMs. Odds are good that it's some delta on my local environment and warrants a refresh.

However, just for _more_ context, `pyright` works as intended when using it as installed by `npm install -g pyright` and fails to resolve, even stdlib packages, when installed inside a poetry-based venv with `poetry add -G dev pyright` and no external packages installed in the venv. May be something to keep an eye out for. 👀
